File: hackerrank/30 days of code/day 15, linked list/day_15_linked_list.js
process.stdin.resume();
process.stdin.setEncoding('ascii');

let input_stdin = "";
let input_stdin_array = "";
let input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin + = data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();    
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

function Node(data) {
    this.data = data;
    this.next = null;
}

function Solution() {

    this.insert = function(head, data) {

        let start
        if (head) {
            start = head
            for ( ; start.next; start = start.next);
            start.next = new Node(data)
        }
        else {
            start = new Node(data)
            head = start
        }

        return head
    }

    this.display = function(head) {

        let start = head;
            while (start) {
                process.stdout.write(start.data + " ");
                start = start.next;
            }
    };
}

function main() {
    let T = +readLine();
    let head = null;
    let mylist = new Solution();

    while (T--)
        head = mylist.insert(head, +readLine());

    mylist.display(head);
}

File: hackerrank/30 days of code/day 15, linked list/day_15_linked_list.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


class Solution:
    def display(self, head):

        current = head
        while current:
            print(current.data, end=" ")
            current = current.next

    def insert(self, head, data):

        if head:
            start = head

            while start.next:
                start = start.next
            start.next = Node(data)

        else:
            start = Node(data)
            head = start

        return head


mylist = Solution()

T = int(input())
head = None

for i in range(T):
    head = mylist.insert(head, int(input()))

mylist.display(head)
